zio-http/shared/src/main/scala/zio/http/Route.scala

```diff
@@ -489,4 +537,16 @@ object Route                   {
     }
   }
 
+  sealed trait IsResponseCauseHandler[-A] { def isResponse: Boolean }
```

Comment:
The most type evidences starting with `Is` tend provide the implicit instance only and if only the type.

As it always provides the implicit value, maybe we need another name to avoid confusion.

Possible alternatives:
- `DetermineResponse`
- `CheckResponse`
- `ResponseTypeChecker`
- ...

I think `CauseHandler` is somewhat misleading as this type is not directly related to handler.

Reply:
Good point. Changing it to CheckResponse